File: Week1/cash.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int coinsCounter = 0;
    double money = -1;
    int coinValues[4] = { 25, 10, 5, 1 };

    while (money < 0)
    {
        money = get_double("How much should the change be?: ");
    }

    int coins = (int)money * 100;
   
    for (int i = 0; i < 4; i++)
    {
        coinsCounter += coins / coinValues[i];
        coins = coins % coinValues[i];
    }

    printf("%i\n", coinsCounter);
}

File: Week5/Sentimental/cash/cash.py
VALUE_TYPES = [25, 10, 5, 1]


def getCountWithGreedy(number):
    operationCounter = 0

    while number:
        for value in VALUE_TYPES:
            operationCounter += number // value
            number %= value

    return operationCounter


def isNumber(str):
    try:
        float(str)
        return True
    except ValueError:
        return False


def main():
    inputText = "How much should the change be?: "
    money = input(inputText)
    while (not isNumber(money)) or float(money) < 0:
        money = input(inputText)

    coinsNeeded = getCountWithGreedy(int(float(money) * 100))
    print(coinsNeeded)

if __name__ == '__main__':
    main()
